frontend/src/metabase-lib/parameters/utils/parameter-type.ts

```diff
@@ -42,9 +41,7 @@ export function isStringParameter(parameter: Parameter) {
   return type === "string";
 }
 
-export function isFieldFilterParameter(
-  parameter: Parameter,
-): parameter is FieldFilterUiParameter {
```

Comment:
This type guard is a lie in isolation. The parameter is really not a `FieldFilterUiParameter` unless it has been transformed by the `buildFieldFilterUiParameter` function. https://github.com/metabase/metabase/blob/c37e7ac9d1213e11fcf0b7639fd4006b84a6abd6/frontend/src/metabase/parameters/utils/dashboards.ts#L132